integration_tests/src/main/python/data_gen.py

```diff
@@ -269,9 +276,9 @@ def __init__(self, precision=None, scale=None, nullable=True, special_cases=None
         super().__init__(DecimalType(precision, scale), nullable=nullable, special_cases=special_cases)
         self.scale = scale
         self.precision = precision
+        self.avoid_positive_values = avoid_positive_values
         negative_pattern = "-" if avoid_positive_values else "-?"
-        self.pattern = negative_pattern + "[0-9]{1,"+ str(precision) + "}e" + str(-scale)
-        self.base_strs = sre_yield.AllStrings(self.pattern, flags=0, charset=sre_yield.CHARSET, max_count=_MAX_CHOICES)
+        self.pattern = negative_pattern + "[0-9]{" + str(precision) + "}e" + str(-scale)
```

Comment:
Why have a pattern at all if we are not going to use it?

Reply:
Done.